compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -1302,15 +1302,19 @@ impl<'a> State<'a> {
                     self.print_visibility(&item.vis);
                     ("macro", false)
                 };
+                let delim = macro_def.body.delim();
                 self.print_mac_common(
                     Some(MacHeader::Keyword(kw)),
                     has_bang,
                     Some(item.ident),
-                    macro_def.body.delim(),
+                    delim,
                     &macro_def.body.inner_tokens(),
                     true,
                     item.span,
                 );
+                if has_bang && matches!(delim, DelimToken::Paren | DelimToken::Bracket) {
```

Comment:
We already have a function for this - `macro_def.body.need_semicolon()`, and `has_bang` should be redundant in this condition.

Reply:
Nice! Thanks for taking a look!